Sources/ForageSDK/Component/ForagePINTextField/Vault/BtPINTextField.swift

```diff
@@ -95,6 +97,22 @@ class BasisTheoryTextFieldWrapper: UIView, VaultWrapper {
                 self._isValid = elementEvent.valid
                 self._isComplete = elementEvent.complete
                 self.delegate?.textFieldDidChange(self)
+            } else if (
+                elementEvent.type == BTEventType.FOCUS.rawValue ||
+                elementEvent.type == BTEventType.BLUR.rawValue
+            ) {
+                // Note: This is custom logic to unblock us while BT makes changes on their
+                // end. They had the incorrect initial state logic for their input field.
+                if (elementEvent.empty) {
+                    self._isEmpty = elementEvent.empty
+                    self._isValid = false
```

Comment:
Probably not too pressuring but in Android, [I had that `isEmpty = true` **is a valid state** but **not a complete state**](https://github.com/teamforage/forage-android-sdk/blob/b9fb8826b47b80f69b45a2eb04d0ce3dafd8d565/forage-android/src/main/java/com/joinforage/forage/android/core/element/state/ElementState.kt#L13). Does anybody have a strong opinion one way or the other? Just want to make sure the functionality is in sync across Android and iOS

Reply:
For the PIN, I `isValid` and `isComplete` both map to checking if the input is of length 4. If we change it, `isValid` would just be true at all times for the PIN right?

Reply:
I need to correct myself because what I said was misleading:
1. the default `isValid` state of an ForageElement is what I would advocate should be `true`
2. on any subsequent update, if the input is found to be empty, then `isValid = false`

In other words, what you have here is fine because I believe this function is the update step. However, a little bit above in this file `isValid` is initialized to false (link [here](https://github.com/teamforage/forage-ios-sdk/blob/8bb9d898da28881940b2031e05e3916507622630/Sources/ForageSDK/Component/ForagePINTextField/Vault/BtPINTextField.swift#L26)). I would advocate that it should be initialized to `true` so that any merchant does not kick things off by displaying an error message of their own because the default state of the `ForageElement` suggest the field is invalid.